Automattic/wp-calypso
# Components: Remove unused ImageSelector and related state

#### Changes proposed in this Pull Request

The `ImageSelector` block was never used since it was created in #26297, almost 3 years ago. This PR removes it, along with its related state that is also not used anywhere.

#### Testing instructions

* Verify that the removed code isn't used anywhere.
* Verify that DropZone continues to work like it did before: `/devdocs/design/drop-zone` or in `/media/:site`.
